src/classes/PS_CommitmentRequest.cls

```diff
@@ -531,8 +567,11 @@ public inherited sharing class PS_CommitmentRequest {
             Date donationDate = new RD2_ScheduleService()
                 .getNextDonationDateGreaterThan(rdSchedule, RD2_ScheduleService.currentDate);
 
-            DateTime donationDatetime = Datetime.newInstance(donationDate.year(), donationDate.month(), donationDate.day());
-            return donationDatetime.format('yyyy-MM-dd');
+            Datetime donationDatetime = donationDate == Datetime.now().date()
+                ? Datetime.now()
+                : Datetime.newInstance(donationDate.year(), donationDate.month(), donationDate.day());
```

Comment:
Since the donation date is calculating from the front end, I need to do a check against the GMT timezone to see if user's today is already GMT's next or previous day. If so, convert the schedule date to GMT time.

Reply:
Is this to prevent a case where `donationDate` is Yesterday when it should be Today?

Reply:
Dang, I would have completely missed this, good to know.